src/plugins/foxPage/hooks/useFarmingApr.ts

```diff
@@ -17,13 +24,55 @@ import { useCurrentBlockNumber } from './useCurrentBlockNumber'
 
 const ethersProvider = getEthersProvider()
 
+function calculateSlippageMargin(percentage: number, amount: string) {
+  const remainingPercentage = (100 - percentage) / 100
+  return bnOrZero(toBaseUnit(amount, 18))
+    .times(bnOrZero(remainingPercentage))
+    .decimalPlaces(0)
+    .toFixed()
+}
+
+function bufferGas(limitOrPrice: string) {
+  return bnOrZero(limitOrPrice).times(1.2).decimalPlaces(0).toFixed()
+}
+
+async function getBufferedGas(tx: TransactionRequest) {
+  let gasLimit = null
+  let gasPrice = null
+  try {
+    if (ethersProvider && tx) {
+      const gas = await ethersProvider?.estimateGas(tx)
+      const price = await ethersProvider?.getGasPrice()
+      if (gas && price) {
+        gasLimit = bufferGas(gas?.toString())
+        gasPrice = bufferGas(price?.toString())
+      }
+    }
+    return {
+      gasLimit,
+      gasPrice,
+    }
+  } catch {
+    throw Error('Problem estimating gas')
+  }
+}
+
+const toHex = (value: BigNumberish) => BigNumber.from(value).toHexString()
```

Comment:
I know this is a reimplementation from the foxfarm repo, but `bnOrZero` will give you a `BN` so you don't need to use `BigNumber.from()`